### ISSUE TITLE
Only show active coupling inputs in inputs summary

### ISSUE THREAD

**Issue by noracato** [quintel/multi-year-charts#76]
A coupled scenario may contain coupled inputs, like the following II3050 scenario. In the ETModel interface we show these coupling inputs by hiding the deactivated slider and showing the coupling icon with the input that was activated by a coupling.

<img width="478" alt="Screenshot 2024-10-11 at 13 19 53" src="https://github.com/user-attachments/assets/27d801ce-4e4e-4954-8205-3343f3f44a5a">

However, the information about if a slider was activated or deactivated by a coupling is not available in the multi-year-charts slider settings interface:

<img width="1453" alt="Screenshot 2024-10-11 at 13 19 47" src="https://github.com/user-attachments/assets/4e602eaa-ac46-48e5-9453-120ea109efcb">

We should show the coupling icon for sliders that were activated by a coupling, and show something (could be another icon) for sliders that were deactivated by a coupling.

**Comment by mabijkerk** [comment]
This has become quite relevant since we allow showing all inputs in the MYC. When that option is selected, all users are confronted by external coupling inputs, which are actually only meant for advanced users. We should only show active inputs - effectively meaning that by default, external coupling inputs should be hidden.

**Comment by louispt1** [comment]
# Background
InputsSummary is where the data is being rendered. The Collections tool uses redux middleware to call the Connection instance conn.fetchInputs() which eventually hits this endpoint in the engine:
`GET /api/v3/scenarios/{scenarioId}/inputs.json?defaults=original`

curl request to have a peek at the data:
```
curl -X GET "https://engine.energytransitionmodel.com/api/v3/scenarios/1335103/inputs" \
  -H "Accept: application/json" \
  -H "Authorization: Bearer $ETM_API_TOKEN"
```

The data comes back like this - so we'll need to parse the coupling_disabled field - coupling_disabled returns if the input is disabled due to a coupling.
```
  "capacity_of_energy_chp_supercritical_waste_mix": {
    "min": 0.0,
    "max": 56356.044687167785,
    "default": 894.6490091116173,
    "unit": "MW",
    "disabled": true,
    "coupling_disabled": true,
    "disabled_by_couplings": [
      "external_model_industry",
      "waste_chp"
    ]
```

you'll notice if the input is not disabled, this field is not set:
```
  "industry_other_oil_in_crude_oil_share": {
    "min": 0.0,
    "max": 100.0,
    "default": 0.0,
    "unit": "%",
    "user": 0.0,
    "disabled": false,
    "disabled_by_couplings": [
      "external_model_industry",
      "industry_oil_mix"
    ],
    "share_group": "industry_crude_oil"
  },
```

# Action plan
- Extend types in utils/api/types.ts to handle this extra data.
- Update InputsSummary/Row.tsx to:
    - Hide disabled inputs.
    - Show active coupling inputs with a chain symbol.

**Comment by mabijkerk** [comment]
This is now occurs again on beta. The slider settings now show coupling inputs again, even though they aren't set. For example, for Steel it now becomes quite messy:

<img width="1604" height="791" alt="Image" src="https://github.com/user-attachments/assets/500efe36-4c17-4387-bd9d-402288f49a0c" />

**Comment by louispt1** [comment]
> This is now occurs again on beta. The slider settings now show coupling inputs again, even though they aren't set. For example, for Steel it now becomes quite messy:
> [<img alt="Image" width="1604" height="791" src="https://private-user-images.githubusercontent.com/67338510/648717904-500efe36-4c17-4387-bd9d-402288f49a0c.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3ODg5NzEzMzcsIm5iZiI6MTc4ODk3MTAzNywicGF0aCI6Ii82NzMzODUxMC82NDg3MTc5MDQtNTAwZWZlMzYtNGMxNy00Mzg3LWJkOWQtNDAyMjg4ZjQ5YTBjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjA5MDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwOTA5VDE2MjM1N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQyMDc4YzNhZTkxODc1OTI3ZTBhODIxMGJjZWZiZTVlZWRkYjdlOWM4YzA1ODJmMDQ3NjE5NmIyZWVhNzI1MDkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JnJlc3BvbnNlLWNvbnRlbnQtdHlwZT1pbWFnZSUyRnBuZyJ9.fTapug4uOs0OJyKz3uUiptltmx9xHHEWHYmTkCflbWM">](https://private-user-images.githubusercontent.com/67338510/648717904-500efe36-4c17-4387-bd9d-402288f49a0c.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3ODg5NjA4NjEsIm5iZiI6MTc4ODk2MDU2MSwicGF0aCI6Ii82NzMzODUxMC82NDg3MTc5MDQtNTAwZWZlMzYtNGMxNy00Mzg3LWJkOWQtNDAyMjg4ZjQ5YTBjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjA5MDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwOTA5VDEzMjkyMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThiNWZkNTYwYzhkZmQzZGU4ZmViMTdjNmIxMjVhMGM1YWFmMjQ2OGMxYzc2ZTgyMzBmYmM0ZjE5MWM0NDI1YjUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JnJlc3BvbnNlLWNvbnRlbnQtdHlwZT1pbWFnZSUyRnBuZyJ9.FyNGqJzMJrUbc44IcDp8BOiG0LBYj0rZ3eh4dWt6quw)

In this case there is an active coupling in two scenarios. The behaviour hasn't changed from before in this respect, only that highlighting the share groups makes it clearer. I believe re-ordering the share group hierarchy will resolve a lot of these cases of 'messiness'.

**Comment by mabijkerk** [comment]
You're right, thanks! I probably made this Collection a while back to test the case with multiple scenarios, where some are coupled and some aren't. It is then indeed a sorting issue, not a technical issue. My bad for reopening this.
